Source/WebCore/html/HTMLInputElement.cpp

```diff
@@ -1532,11 +1533,21 @@ URL HTMLInputElement::src() const
     return document().completeURL(attributeWithoutSynchronization(srcAttr));
 }
 
+void HTMLInputElement::logUserInteraction()
+{
+    RefPtr mainFrame = document().frame() && document().frame()->localMainFrame() ? document().frame()->localMainFrame() : nullptr;
+    if (RefPtr document = mainFrame ? mainFrame->document() : nullptr)
+        ResourceLoadObserver::shared().logUserInteractionWithReducedTimeResolution(*document);
```

Comment:
I'd have written this with early returns, rather than using `nullptr` twice to drive the control flow. It would also remove the duplication of `document().frame()` on the first line.

Reply:
Thanks, made those changes.